test_integration/test_plugin_static_policy.py

```diff
@@ -0,0 +1,115 @@
+from test_integration import util
+import geopmpy.topo
+
+# todo this function results in test launcher
+# invocation, and is called 3 times (1 by class and
+# in 2 decorator tests).  may be useful to add once logic
+def geopmadmin():
+    settings = {}
+    for option in ["--config-default", "--config-override"]:
+        try:
+            proc = subprocess.Popen(shlex.split("geopmadmin {}".format(option)),
+                                    stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            config_file = proc.stdout.readline()
+            stdout = io.StringIO()
+            stderr = io.StringIO()
+            geopm_test_launcher.allocation_node_test('dummy -- cat {}'.format(config_file),
+                                                     stdout, stderr)
+            settings.update(json.loads(stdout.getvalue()))
+        except subprocess.CalledProcessError:
+            # config_file may not exist and will cause allocation_node_test to fail
+            pass
+    return settings
+
+def geopmadminagent():
+    geopm_system_config = geopmadmin()
+    return geopm_system_config['GEOPM_AGENT']
+
+def geopmadminagentpolicy():
+    geopm_system_config = geopmadmin()
+    stdout = io.StringIO()
+    stderr = io.StringIO()
+    geopm_test_launcher.allocation_node_test('dummy -- cat {}'.format(geopm_system_config['GEOPM_POLICY']),
+                                             stdout, stderr)
+    return json.loads(stdout.getvalue())
+
+class TestIntegrationPluginStaticPolicy(unittest.TestCase):
+    """Test the static policy enforcement feature of the currently
+       configured RM plugin.
+
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls._tmp_files = []
+        cls._geopmadminagent = geopmadminagent()
+        cls._geopmadminagentpolicy = geopmadminagentpolicy()
+
+    @unittest.skipUnless(geopmadminagent() in ['frequency_map', 'energy_efficient'],
+                         'Requires environment defaut/override to be configured to cap frequency.')
+    def test_frequency_cap_enforced(self):
+        if self._geopmadminagent == 'frequency_map':
+            test_freq = self._geopmadminagentpolicy['FREQ_MAX']
+        elif self._geopmadminagent == 'energy_efficient':
+            test_freq = self._geopmadminagentpolicy['FREQ_FIXED']
+        current_freq = geopm_test_launcher.geopmread("MSR::PERF_CTL:FREQ board 0")
+        self.assertEqual(test_freq, current_freq)
+
+    @unittest.skipUnless(geopmadminagent() in ['power_governor','power_balancer'],
+                         'Requires environment defaut/override to be configured to cap power.')
+    def test_power_cap_enforced(self):
+        test_power = self._geopmadminagentpolicy['POWER_PACKAGE_LIMIT_TOTAL']
+        # todo need test_launcher to execute topo on compute node
+        num_pkg = geopmpy.topo.num_domain('package')
+        current_power = num_pkg * geopm_test_launcher.geopmread("MSR::PKG_POWER_LIMIT:PL1_POWER_LIMIT board 0")
```

Comment:
It isn't safe to assume that the PL1 limit is the same on both sockets if multisocket.

I was going to suggest you use num_pkg to loop over the packages, and use geopmread to read the limit on the individual sockets.  But I see you're reading the board limit...

I think this should fail:  reading the board limit should read each package and sum them I think.  There's a bug somewhere here, I think it's in how geopmread sums (or not) the values for the board domain if this logic works.

Reply:
inherited from here: https://github.com/geopm/geopm/blob/dev/test_integration/test_static_policy.py#L173-L180
shall we file a bug?

Reply:
There is a subtlety here that we both missed.  The policy name is: POWER_PACKAGE_LIMIT_TOTAL, as in total for all of the packages on the board.  That's why sampling board works but yes this implies they packages split the allotted power equally.  Is a non uniform distribution of power across packages expected to appear in the wild?